src/components/App/App.js

```diff
@@ -3,29 +3,61 @@ import { useEffect, useState } from "react";
 import { Route, Routes } from "react-router-dom";
 import "./App.css";
 import { Main } from "../Main/Main";
-import { SavedNews } from "../SavedNews/SavedNews";
 import { Layout } from "../Layout/Layout";
 import { useMediaQuery } from "react-responsive";
 import { Register } from "../Register/Register";
-import { Login } from "../Login/Login";
 import { InfoToolTip } from "../InfoToolTip/InfoToolTip";
 import { PopupWithForm } from "../PopupWithForm/PopupWithForm";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import * as auth from "../../utils/auth";
+import { ProtectedRoute } from "../ProtectedRoute/ProtectedRoute";
+import FormContextProvider from "../../contexts/FormContext";
+import { Login } from "../Login/Login";
+import { mainApi } from "../../utils/MainApi";
+import { CardsContext } from "../../contexts/SavedCardsContext";
+import { SavedNews } from "../SavedNews/SavedNews";
 
 function App() {
   const isMonitorOrTablet = useMediaQuery({ minWidth: 768 });
   const isMonitor = useMediaQuery({ minWidth: 1024 });
   const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 1023 });
   const isMobile = useMediaQuery({ maxWidth: 767 });
 
+  const [currentUser, setCurrentUser] = useState({});
+  const [keyword, setKeyword] = useState("");
+  const [showArticles, setShowArticles] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
   const [isLoggedIn, setIsLoggedIn] = useState(false);
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-  const [userName, setUserName] = useState("");
   const [isRegistered, setIsRegistered] = useState(false);
   const [isDropdownMenuOpen, setIsDropdownMenuOpen] = useState(false);
   const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
   const [isLoginPopupOpen, setIsLoginPopupOpen] = useState(false);
   const [isInfoToolTipOpen, setIsInfoToolTipOpen] = useState(false);
+  const [startIndex, setStartIndex] = useState(0);
+  const [endIndex, setEndIndex] = useState(3);
+  const [totalResult, setTotalResult] = useState(0);
+
+  const [savedCards, setSavedCards] = useState([]);
+  const cardsData = [savedCards, setSavedCards];
+
+  useEffect(() => {
+    if (localStorage.getItem("token") !== null) {
+      localStorage.getItem("token");
+      auth
+        .getContent()
+        .then(({ user }) => {
+          setIsRegistered(true);
+          setIsLoggedIn(true);
+          setCurrentUser(user);
+        })
+        .then(
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the reference to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server.


```js

        .then(() => {
          mainApi
            .getSavedArticles()
            .then((data) => {
              setSavedCards(data);
            })
            .catch((err) => console.log(`Error..... ${err}`));
        })
        .catch((err) => console.log(`Error..... ${err}`));

```